of-builder/main.go

```diff
@@ -88,17 +92,50 @@ func build(w http.ResponseWriter, r *http.Request) ([]byte, error) {
 	if cfg.Frontend == "" {
 		cfg.Frontend = "tonistiigi/dockerfile:v0"
 	}
-
-	cmd := exec.Command("buildctl", "--debug", "build", "--frontend=gateway.v0", "--frontend-opt=source="+cfg.Frontend, "--local=context="+filepath.Join(tmpdir, "context"), "--local=dockerfile="+filepath.Join(tmpdir, "context"), "--no-progress", "--exporter=image",
-		"--exporter-opt=name="+cfg.Ref, "--exporter-opt=push=true", "--exporter-opt=registry.insecure=true")
-	env := os.Environ()
-	env = append(env, "BUILDKIT_HOST=tcp://of-buildkit:1234")
-	cmd.Env = env
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-	if err := cmd.Run(); err != nil {
+	contextDir := filepath.Join(tmpdir, "context")
+	solveOpt := client.SolveOpt{
+		Exporter: "image",
+		ExporterAttrs: map[string]string{
+			"name":              cfg.Ref,
+			"push":              "true",
+			"registry.insecure": "true",
+		},
+		LocalDirs: map[string]string{
+			"context":    contextDir,
+			"dockerfile": contextDir,
+		},
+		Frontend: "gateway.v0",
+		FrontendAttrs: map[string]string{
+			"source": cfg.Frontend,
+		},
+		// ~/.docker/config.json could be provided as Kube or Swarm's secret
+		Session: []session.Attachable{authprovider.NewDockerAuthProvider()},
+	}
+	c, err := client.New("tcp://of-buildkit:1234", client.WithBlock())
```

Comment:
This address should probably move to an env-var for configuration. @johnmccabe how did you come to pick 1234 as a port in your original PR? (is this a convention for gRPC?)

Reply:
Its buildkits default, see [here](https://github.com/moby/buildkit/tree/896e65bd0978ae3b74e78d7ae8f19ef6d906740c#running-containerized-buildkit).